Cannot build with Visual Studio 2017

When attempting to build a project with the Antlr4 package installed, the following error is reported.

```text
MSBUILD : error MSB4166: Child node "2" exited prematurely. Shutting down. Diagnostic information may be found in files in the temporary files directory named MSBuild_*.failure.txt.
```

Additional information is given in the MSBuild failure log:

```text
UNHANDLED EXCEPTIONS FROM PROCESS 47004:
=====================
1/12/2017 4:49:20 PM
System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Build.Framework, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'Microsoft.Build.Framework, Version=15.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Build.Framework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.
File name: 'Microsoft.Build.Framework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'


WRN: Assembly binding logging is turned OFF.
To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1.
Note: There is some performance penalty associated with assembly bind failure logging.
To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].

   at Antlr4.Build.Tasks.AntlrClassGenerationTaskInternal.HandleErrorDataReceived(Object sender, DataReceivedEventArgs e)
   at System.Diagnostics.Process.ErrorReadNotifyUser(String data)
   at System.Diagnostics.AsyncStreamReader.FlushMessageQueue()
   at System.Diagnostics.AsyncStreamReader.ReadBuffer(IAsyncResult ar)
   at System.IO.Stream.ReadWriteTask.InvokeAsyncCallback(Object completedTask)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.IO.Stream.ReadWriteTask.System.Threading.Tasks.ITaskCompletionAction.Invoke(Task completingTask)
   at System.Threading.Tasks.Task.FinishContinuations()
   at System.Threading.Tasks.Task.FinishStageThree()
   at System.Threading.Tasks.Task.FinishStageTwo()
   at System.Threading.Tasks.Task.Finish(Boolean bUserDelegateExecuted)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)
   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)
   at System.Threading.Tasks.Task.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

--- a/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTask.cs
+++ b/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTask.cs
@@ -22,8 +22,6 @@ namespace Antlr4.Build.Tasks
     public class Antlr4ClassGenerationTask
         : Task
     {
-        private static AppDomain _sharedAppDomain;
-
         private const string DefaultGeneratedSourceExtension = "g4";
         private List<ITaskItem> _generatedCodeFiles = new List<ITaskItem>();
 
@@ -163,27 +161,8 @@ namespace Antlr4.Build.Tasks
             }
         }
 
-        [SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate")]
-        public AppDomain GetAntlrTaskAppDomain()
-        {
-            if (_sharedAppDomain != null)
-                return _sharedAppDomain;
-
-            AppDomainSetup info = new AppDomainSetup
-            {
-                ApplicationBase = BuildTaskPath,
-                LoaderOptimization = LoaderOptimization.MultiDomainHost,
-                ShadowCopyFiles = "true"
-            };
-
-            string friendlyName = "AntlrClassGenerationDomain_" + Guid.NewGuid();
-            _sharedAppDomain = AppDomain.CreateDomain(friendlyName, AppDomain.CurrentDomain.Evidence, info, new NamedPermissionSet("FullTrust"), new StrongName[0]);
-            return _sharedAppDomain;
-        }
-
         public override bool Execute()
         {
-            AppDomain domain = null;
             bool success;
 
             if (!Path.IsPathRooted(ToolPath))
@@ -194,8 +173,7 @@ namespace Antlr4.Build.Tasks
 
             try
             {
-                domain = GetAntlrTaskAppDomain();
-                AntlrClassGenerationTaskInternal wrapper = CreateBuildTaskWrapper(domain);
+                AntlrClassGenerationTaskInternal wrapper = CreateBuildTaskWrapper();
                 success = wrapper.Execute();
 
                 if (success)
@@ -215,11 +193,6 @@ namespace Antlr4.Build.Tasks
 
                 ProcessExceptionAsBuildMessage(exception);
                 success = false;
-            }
-            finally
-            {
-                if (domain != null && domain != _sharedAppDomain)
-                    AppDomain.Unload(domain);
             }
 
             return success;
@@ -261,9 +234,9 @@ namespace Antlr4.Build.Tasks
             }
         }
 
-        private AntlrClassGenerationTaskInternal CreateBuildTaskWrapper(AppDomain domain)
+        private AntlrClassGenerationTaskInternal CreateBuildTaskWrapper()
         {
-            AntlrClassGenerationTaskInternal wrapper = (AntlrClassGenerationTaskInternal)domain.CreateInstanceAndUnwrap(Assembly.GetExecutingAssembly().FullName, typeof(AntlrClassGenerationTaskInternal).FullName);
+            AntlrClassGenerationTaskInternal wrapper = new AntlrClassGenerationTaskInternal();
 
             IList<string> sourceCodeFiles = null;
             if (this.SourceCodeFiles != null)

--- a/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTask.cs
+++ b/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTask.cs
@@ -6,11 +6,8 @@ namespace Antlr4.Build.Tasks
     using System;
     using System.Collections.Generic;
     using System.Diagnostics;
-    using System.Diagnostics.CodeAnalysis;
     using System.Linq;
     using System.Reflection;
-    using System.Security;
-    using System.Security.Policy;
     using System.Threading;
     using Microsoft.Build.Framework;
     using Microsoft.Build.Utilities;

--- a/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTaskInternal.cs
+++ b/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTaskInternal.cs
@@ -10,9 +10,9 @@ namespace Antlr4.Build.Tasks
     using System.Linq;
     using System.Reflection;
     using System.Text.RegularExpressions;
-    using RegistryHive = Microsoft.Win32.RegistryHive;
     using RegistryKey = Microsoft.Win32.RegistryKey;
 #if NET40PLUS
+    using RegistryHive = Microsoft.Win32.RegistryHive;
     using RegistryView = Microsoft.Win32.RegistryView;
 #else
     using Registry = Microsoft.Win32.Registry;

--- a/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTaskInternal.cs
+++ b/runtime/CSharp/Antlr4BuildTasks/Antlr4ClassGenerationTaskInternal.cs
@@ -19,7 +19,7 @@ namespace Antlr4.Build.Tasks
 #endif
     using StringBuilder = System.Text.StringBuilder;
 
-    internal class AntlrClassGenerationTaskInternal : MarshalByRefObject
+    internal class AntlrClassGenerationTaskInternal
     {
         private List<string> _generatedCodeFiles = new List<string>();
         private IList<string> _sourceCodeFiles = new List<string>();

